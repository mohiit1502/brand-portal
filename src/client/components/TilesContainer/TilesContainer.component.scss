.c-TilesContainer {
  padding: 1.5rem 2.5rem;
//   width: 100%;
}
