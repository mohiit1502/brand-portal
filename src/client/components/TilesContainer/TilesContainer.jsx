import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import Tile from "./../Tile";
import ContentRenderer from "./../../utility/ContentRenderer";
import "./TilesContainer.component.scss";

const TilesContainer = props => {

  const [contentRenderer, setContentRenderer] = useState();

  useEffect(() => {
    if (!contentRenderer) {
      setContentRenderer(new ContentRenderer());
    }
  }, []);

  const tilesRenders = contentRenderer && props.tiles && props.tiles.map((tile, key) => <Tile key={key} data={tile} contentRenderer={contentRenderer} />);

  return (
    <div className="c-TilesContainer row display-flex px-4 mx-0">
      {tilesRenders}
    </div>
  );
};

TilesContainer.propTypes = {
  tiles: PropTypes.array
};

export default TilesContainer;
