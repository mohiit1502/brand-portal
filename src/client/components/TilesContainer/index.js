import TilesContainer from './TilesContainer.jsx';

export default TilesContainer;