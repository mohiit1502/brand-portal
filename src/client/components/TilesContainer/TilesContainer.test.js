import React from 'react';
import TilesContainer from './TilesContainer';

describe('TilesContainer', () => {
   it('renders without error', () => {

   });
});