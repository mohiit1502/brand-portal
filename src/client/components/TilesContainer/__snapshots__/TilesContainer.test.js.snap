// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TilesContainer renders without error 1`] = `
<div
  className="c-TilesContainer row display-flex mx-0"
/>
`;
